### DISCUSSION TITLE
New ShapeKernel examples: MeshPainter, ImplicitGenus and ImplicitSuperEllipsoid

### DISCUSSION THREAD

**Discussion by josefinelissner** [leap71/PicoGK#22] [Show and tell]
Our ShapeKernel library got three new examples today, two of which where contributed by @AasimRashid

1) MeshPainter is a new class that splits a mesh into sub-meshes according to custom properties that can be defined via a delegate function. Each sub-mesh is then colored individually according to a color scale objects and so you can visualize properties of an object in 3D. The obvious example is to "paint" the local overhang angles onto the shape. Run `Leap71.ShapeKernelExamples.MeshPainterShowCase.Task` to check it out

<img width="938" alt="Screenshot 2024-03-15 at 3 25 37 PM" src="https://github.com/leap71/PicoGK/assets/44703282/198458f9-3a3d-4585-8cd8-b927bd64d34e">

2) Run `Leap71.ShapeKernelExamples.ImplicitSolidEllipsoid.Task` to generate three versions of a SuperEllipsoid... and find more possible parameter sets here: https://en.wikipedia.org/wiki/Superellipsoid

<img width="1052" alt="Screenshot 2024-03-15 at 3 07 12 PM" src="https://github.com/leap71/PicoGK/assets/44703282/8bce81dd-88b0-44ff-9a9e-60c350b9581e">

3) Run `Leap71.ShapeKernelExamples.ImplicitGenusGyroid.Task` to get an example of an "implicit surface of genus 2" (https://en.wikipedia.org/wiki/Implicit_surface) combined with an implicit gyroid pattern:

<img width="1035" alt="Screenshot 2024-03-15 at 3 08 30 PM" src="https://github.com/leap71/PicoGK/assets/44703282/4a89c1b2-772b-4093-810f-4d6108c1be48">

**Comment by josefinelissner** [comment]
Here is the MeshPainter at work on some more elaborate shapes:

<img width="575" alt="Screenshot 2024-03-15 at 3 25 27 PM" src="https://github.com/leap71/PicoGK/assets/44703282/15e75753-c01f-4350-ba40-370a9fcef930">

<img width="776" alt="Screenshot 2024-03-15 at 3 27 51 PM" src="https://github.com/leap71/PicoGK/assets/44703282/bc2801ca-3d8f-4190-8faa-5fa10b9fcad0">

**Reply by rawwerks** [reply]
the critical overhangs example is so cool! very powerful
